### DISCUSSION TITLE
Invalid languages is shown in GitHub lang stats

### DISCUSSION THREAD

**Discussion by Demianeen** [anuraghazra/github-readme-stats#3730] [Q&A] [answered]
Hi! I really like your plugin, so I wanted to try it out. But I met the issue that it displays some languages I never used before: c# and ShaderLab (what is it😂). Take a look:

![CleanShot 2024-04-17 at 14 18 53@2x](https://github.com/anuraghazra/github-readme-stats/assets/51330172/4586f054-934b-4a31-8ff6-fdc06711b886)

I don't have any c# repos:

![CleanShot 2024-04-17 at 14 23 51@2x](https://github.com/anuraghazra/github-readme-stats/assets/51330172/5de33a35-6877-4551-ae3d-08d51a2970c6)

I didn't even have ShaderLab in languages:

![CleanShot 2024-04-17 at 14 24 25@2x](https://github.com/anuraghazra/github-readme-stats/assets/51330172/e3ade279-e2f5-4645-a230-8f5106a6bf61)

Why did this happen? And what should I do?

The URL I used:
https://github-readme-stats.vercel.app/api/top-langs?username=demianeen&locale=en&hide_title=false&card_width=320&langs_count=5&hide_border=false

**Comment by qwerty541** [comment, marked as answer]
Hey, @Demianeen. Actually you own this repository https://github.com/Demianeen/mafia-3D which contains C# and ShaderLab languages. Before writing that something doesn't work you could more properly explore your repositories. If you don't want this repository to be included in your stats card, then you can use `&exclude_repo=mafia-3D` query string option. Here is an example:

```html
https://github-readme-stats.vercel.app/api/top-langs?username=demianeen&locale=en&hide_title=false&card_width=320&langs_count=5&hide_border=false&exclude_repo=mafia-3D
```

<img src="https://github-readme-stats.vercel.app/api/top-langs?username=demianeen&locale=en&hide_title=false&card_width=320&langs_count=5&hide_border=false&exclude_repo=mafia-3D" />

**Reply by iibrahemali** [reply]
That was I am tryna say abt calculations of the top langs. Cuz I have just added one HTML file, it counts it as 80% used lang.

**Reply by qwerty541** [reply]
By default languages percentages are calculated by bytes count, so it's an expected behaviour. However you are able to configure how languages calculation algorithm works, there is an ability to count languages usage based on count of repositories where the particular languages was used, also you can combine these two languages usage calculation strategies. To configure your languages calculation algorithm you should use the following query string parameters: `&size_weight=(0-1.0)` and `&count_weight=(0-1.0)`. You can find more information in our documentation, highly recommend to read: https://github.com/anuraghazra/github-readme-stats?tab=readme-ov-file#language-stats-algorithm
